packages/core/src/components/forms/inputGroup.tsx

```diff
@@ -106,17 +99,15 @@ export class InputGroup extends React.Component<HTMLInputProps & IInputGroupProp
         );
     }
 
-    private updateInputWidth() {
-        if (this.rightElement != null) {
-            const { clientWidth } = this.rightElement;
-            // small threshold to prevent infinite loops
-            if (Math.abs(clientWidth - this.state.rightElementWidth) > 2) {
-                this.setState({ rightElementWidth: clientWidth });
-            }
-        } else {
-            this.setState({ rightElementWidth: 0 });
-        }
-    }
+    private handleOnBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        this.props.onBlur != null ? this.props.onBlur(e) : undefined;
+        this.setState({ isActive: !this.state.isActive });
+    };
+
+    private handleOnFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+        this.props.onFocus != null ? this.props.onFocus(e) : undefined;
+        this.setState({ isActive: !this.state.isActive });
```

Comment:
should probably be explicit: set to `true` here and `false` in blur, since we know the desired end state and it's possible that multiple quick state transitions could break this.

Reply:
yep

Reply:
@giladgray stashing this and punting for now